### PR TITLE
5th Assignment: Added Search Functionality for our Team G - Gladiators

### PR DESCRIPTION
**Pull request by nitishnivedan**
5th Assignment: Added Search Functionality for our Team G - Gladiators

### CODE REVIEW

File: src/Assignment4/CardComponent/index.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
 import React from "react";
 
+import userIcon from "../../logo/userIcon.png";
 import "./styles-card.css";
 
 const CardComponent = ({ temmatesDetails }) => (
   <div className="CardWrapper">
-    <img src={temmatesDetails.imageSource} className="imageClass" />{" "}
+    <img
```
Comment (ap221882) [inline src/Assignment4/CardComponent/index.js]: Can destructure it more like this: 
{temmatesDetails: { name, location, company}}

File: src/Assignment4/Root/index.js [hunk 2]
```diff
@@ -1,96 +1,23 @@
...
     <>
       <Header />
-      {teamInformation &&
-        teamInformation.length > 0 &&
-        teamInformation.map((temmatesDetails) => (
+      <SearchComponent
+        setCurrentCards={setCurrentCards}
+        teamInformation={teamInformation}
+      />
+      {currentCards &&
```
Comment (ap221882) [inline src/Assignment4/Root/index.js:18]: Will not it be more intuitive and readable to make a separate Cards component and pass the prop into it? Here you're passing the copy of currrentCards' state's single element to the CardComponent. Any theory or coding style behind it?

File: src/Assignment4/SearchComponent/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import "./styles.css";
+
+import { useState } from "react";
```
Comment (ap221882) [inline src/Assignment4/SearchComponent/index.js:1]: Imports from global modules should be on the top. Isn't it?

**Review by ap221882 — COMMENTED** [review]
Well written and concise. Added some suggestions and doubts.

File: src/Assignment4/CardComponent/index.js (hunk 1 above)
Comment (nitishnivedan) [reply src/Assignment4/CardComponent/index.js]: good one. thanks

File: src/Assignment4/SearchComponent/index.js (hunk 3 above)
Comment (nitishnivedan) [reply src/Assignment4/SearchComponent/index.js:1]: done

File: src/Assignment4/Root/index.js (hunk 2 above)
Comment (nitishnivedan) [reply src/Assignment4/Root/index.js:18]: didnt get the context. I already have a separate Card Component and passing props to it.  Any link or something can you share that will help me to understand.

Comment (ap221882) [reply src/Assignment4/Root/index.js:18]: My bad. No need for Cards component.

File: src/Assignment4/SearchComponent/index.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import "./styles.css";
+
+const SearchComponent = ({ setCurrentCards, teamInformation }) => {
+  const [nameOrDesignation, setNameOrDesignation] = useState("");
+  return (
+    <form
+      onSubmit={(event) => {
```
Comment (shaksham08) [inline src/Assignment4/SearchComponent/index.js:9]: Rathe than writing this long piece of code to handle on submit you can also make a function like handleSubmit and attach it here, just a suggestion as it Looks good and more readable.

**Review by shaksham08 — APPROVED** [review]
Just added one comment , overall looks good to me

File: src/Assignment4/SearchComponent/index.js (hunk 4 above)
Comment (nitishnivedan) [reply src/Assignment4/SearchComponent/index.js:9]: sure, i generally do that. will push it in my next PR.

**Review by iswarya15 — APPROVED** [review]
Reviewed. All looks good. As Shaksham suggested, it would make code more readable if there is a handlesubmit method..

**Review by smriti-porwal — APPROVED** [review]

**Comment by smriti-porwal** [comment]
can split onSubmit(), other than that everything look good.

**Pull request merged**